### PR TITLE
[WIP] refactor: use etcher-sdk

### PR DESCRIPTION
**Pull request by zvin**
[Etcher SDK](https://github.com/resin-io-modules/etcher-sdk)

TODO:

- [x] lzma support
- [x] gzip support
- [x] bzip2 support
- [x] respect the "Auto-unmount on success" option
- [x] support `manifest.json` files in archives
- [x] use `etcher-sdk` in `etcher-cli`
- [x] move sdk tests to `etcher-sdk`
- [x] remove `lib/sdk`
- [x] handle destinations `open` errors
- [x] raspberry pi usbboot progress bar
- [x] show the error screen when all destinations fail
- [x] merge https://github.com/resin-io-modules/node-raspberrypi-usbboot/pull/10
- [x] fix shrinkwrap
- [x] fix missing "mime" dependency when packaged
- [x] Fix invalid state percentage when flashing compressed images https://github.com/resin-io-modules/etcher-sdk/issues/10
- [x] driverless device adapter (needs testing: https://github.com/resin-io-modules/etcher-sdk/issues/16)
- [x] fix ext2fs electron windows builds & update to latest https://github.com/resin-io/node-ext2fs/issues/34
- [x] fix flash-after-verify bug https://github.com/resin-io-modules/etcher-sdk/issues/12
- [x] fix block / chunk size for compressed images https://github.com/resin-io-modules/etcher-sdk/issues/3
- [x] fix "Error: Could not initialize libusb. Check that your system has a usb controller." https://github.com/resin-io-modules/etcher-sdk/issues/14
- [ ] fix DEBUG env var not being propagated through to the etcher-sdk
- [x] lzma-native attempts to load wrong binding under Electron; https://github.com/resin-io-modules/etcher-sdk/issues/15
- [x] etcher-sdk's build / install step doesn't clean `build` directory, potentially causing stale, incorrect requires https://github.com/resin-io-modules/etcher-sdk/issues/7
- [x] Fix "unsupported image-type .img.gz" https://github.com/resin-io-modules/etcher-sdk/issues/8
- [x] Fix https://github.com/resin-io-modules/drivelist/pull/302
- [x] release etcher-sdk on npm https://github.com/resin-io-modules/etcher-sdk/issues/13
- [x] fix EINVAL on macOS with gzipped images

### CODE REVIEW

File: lib/shared/units.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ const prettyBytes = require('pretty-bytes')
  * @description
  * 1 MB = 1e+6 B
```
Comment (lurch) [inline lib/shared/units.js]: Update the description too?

Comment (lurch) [reply lib/shared/units.js]: Does #1669 help?

Comment (zvin) [reply lib/shared/units.js]: Will do, I need to replace prettyBytes with something else too.Actually I reverted this as it is unrelated to the use of the sdk.

Comment (zvin) [reply lib/shared/units.js]: Actually I reverted this as it is unrelated to the use of the sdk.

**Comment by thundron** [comment]
@resin-ci retest

**Comment by thundron** [comment]
@resin-ci retest

**Comment by zvin** [comment]
@resin-ci retest

**Comment by zvin** [comment]
@resin-ci retest

**Comment by thundron** [comment]
@resin-ci retest

**Comment by thundron** [comment]
@resin-ci retest

**Pull request merged**